cmd/gapit/memory.go

```diff
@@ -0,0 +1,293 @@
+			Index: uint32(mem.AllocationFlagsIndex),
+		}).Path())
+		if err != nil {
+			return log.Errf(ctx, err, "Failed to load allocation flag names")
+		}
+		constants := boxedConstants.(*service.ConstantSet)
+		// If not a bitfield, we can't compare it against the flags
+		if constants.IsBitfield {
+			allocationFlags = constants.Constants
+		}
+	}
+
+	fmt.Printf("%v memory allocations\n", len(mem.Allocations))
+	sort.Slice(mem.Allocations, func(i, j int) bool {
+		return mem.Allocations[i].Handle < mem.Allocations[j].Handle
+	})
+	for _, alloc := range mem.Allocations {
+		fmt.Println("Name:", alloc.Name)
+		fmt.Println("\tDevice:      ", alloc.Device)
+		fmt.Println("\tMemory Type: ", alloc.MemoryType)
+		fmt.Println("\tSize:        ", alloc.Size)
+
+		if alloc.Flags != 0 && len(allocationFlags) != 0 {
+			fmt.Println("\tFlags:")
+			for _, f := range allocationFlags {
+				if (alloc.Flags & uint32(f.Value)) != 0 {
+					fmt.Printf("\t\t%v\n", f.Name)
+				}
+			}
+		}
+
+		if alloc.Mapping.Size != 0 {
+			fmt.Printf("\tMapped into host memory at 0x%x\n",
+				alloc.Mapping.MappedAddress)
+			fmt.Println("\t\tOffset:", alloc.Mapping.Offset)
+			fmt.Println("\t\tSize:  ", alloc.Mapping.Size)
+		}
+
+		bindings := bindingSlice(alloc.Bindings)
+		sort.Slice(bindings, bindings.bindingLess)
+		fmt.Printf("\t%v bindings:\n", len(bindings))
+		for _, binding := range bindings {
+			var typ string
+			switch binding.Type.(type) {
+			case *api.MemoryBinding_Buffer:
+				typ = "Buffer"
+			case *api.MemoryBinding_Image:
+				typ = "Image"
+			case *api.MemoryBinding_SparseImageBlock:
+				typ = "Sparse Image Block"
+			case *api.MemoryBinding_SparseImageMetadata:
+				typ = "Sparse Image Metadata"
+			case *api.MemoryBinding_SparseImageMipTail:
+				typ = "Sparse Image Mip Tail"
+			case *api.MemoryBinding_SparseOpaqueImageBlock:
+				typ = "Sparse Opaque Image Block"
+			case *api.MemoryBinding_SparseBufferBlock:
+				typ = "Sparse Buffer Block"
+			}
+			fmt.Printf("\t%v: %v\n", typ, binding.Name)
+
+			fmt.Println("\t\tOffset:", binding.Offset)
+			fmt.Println("\t\tSize:  ", binding.Size)
+
+			stringifyAspects := func(aspects []api.AspectType) string {
+				if len(aspects) == 0 {
+					return ""
+				}
+				strs := make([]string, len(aspects))
+				for i, asp := range aspects {
+					var typ string
+					switch asp {
+					case api.AspectType_COLOR:
+						typ = "Color"
+					case api.AspectType_DEPTH:
+						typ = "Depth"
+					case api.AspectType_STENCIL:
+						typ = "Stencil"
+					}
+					strs[i] = typ
+				}
+				return strings.Join(strs, ", ")
+			}
+			switch val := binding.Type.(type) {
+			case *api.MemoryBinding_SparseImageBlock:
+				info := val.SparseImageBlock
+				fmt.Printf("\t\tBlock Offset: (%v, %v)\n",
+					info.XOffset, info.YOffset)
+				fmt.Printf("\t\tBlock Extent: (%v, %v)\n",
+					info.Width, info.Height)
+				fmt.Printf("\t\tMip Level:    %v\n", info.MipLevel)
+				fmt.Printf("\t\tArray Layer:  %v\n", info.ArrayLayer)
```

Comment:
Nit: There is the [`text/tabwriter`](https://golang.org/pkg/text/tabwriter/) package that does automatic alignment of text into columns. It also deals with emitting whitespace instead of tab characters (which many terminals will struggle with).